### ISSUE TITLE
Is there an easy way to delete a track?

### ISSUE THREAD

**Issue by ghost** [naomiaro/waveform-playlist#32]
Hi, 
First of all, thanks for this project, it is exactly what I searched for weeks now.

I wonder if there is an easy way to delete a single track? It doesn't seem to have a specific method for the same.  Maybe by getting the data of the track to delete (getTrackDetails() ?) and by removing it from the complete playlist details?

Kind regards,

Emmanuel

**Comment by naomiaro** [comment]
I haven't currently implemented any way to delete a track, so this would have to be a new feature.

**Comment by ezimmert** [comment]
Hi Naomi,

I managed to delete a single track by adding a close button with _deletetrack_ emitter on click in track controls and 
```
ee.on('deletetrack', function (track) {
	var trackInfo = JSON.stringify(track.getTrackDetails());
	var tracksInfo = JSON.stringify(playlist.getInfo());
	var newPlaylist = tracksInfo.replace(trackInfo, '').replace(',,', ',');
	if ( newPlaylist.charAt(newPlaylist.length - 2) == ',' ) {
		newPlaylist = newPlaylist.slice(0,-2) + ']';
	}
	if ( newPlaylist.charAt(1) == ',' ) {
		newPlaylist = '[' + newPlaylist.slice(2, newPlaylist.length);
	}
	ee.emit('clear');
	playlist.soloedTracks = [];
	playlist.mutedTracks = [];
	playlist.load(JSON.parse(newPlaylist));
});
```

It's not very pretty but it works.

**Comment by ptfly** [comment]
One solution that seems to work for now, until the author releases something official:
```
// this.daw = waveform-playlist instance

deleteTrack(index)
{
	this.daw.tracks = this.daw.tracks.filter((track, i) => {
	    if(i === index){
	        track.scheduleStop();
	    }
	    return i !== index
	});

	this.daw.adjustDuration();
	this.daw.draw(this.daw.render());
}
```

@naomiaro Thanks for the awesome work here!

**Comment by naomiaro** [comment]
Thanks @ptfly ! Yes I should put something together soon to handle deletion and reordering.

**Comment by ptfly** [comment]
It'll be awesome, this projects needs few more things and will become a great base for a web based DAW.
Reordering, deletion, adding new empty tracks and the ability to set the source the these tracks afterwards. Chain some FX per track as well as the master out. I guess the masted could be done by supplying my own audio context. 

My plan is to try to connect pedalboard.js audio context and see what will come out from that :)

**Comment by pbaveja** [comment]
Great work on this project @naomiaro.
Here's how I implemented **deleting a single track**.

1. Added a button next to _mute/solo_ in `Track.js`:

```
h(`span.mbtn-gray-custom.btn.btn-default.btn-xs.btn-deleteTrack`, {
            onclick: () => {
              this.ee.emit('deleteTrack', this);
            },
          }, ['Detele Track']),
```

2.  Inside `Playlist.js`, adding an event listener + main deleting func: 

```
ee.on('deleteTrack', (track) => {
      this.deleteTrack(track);
      this.adjustTrackPlayout();
      this.drawRequest();
    });
```

3. And by slightly modifying @ptfly's answer to fit directly:

```
deleteTrack(track) {
    const index = this.tracks.indexOf(track);
    this.tracks = this.tracks.filter((track, i) => {
      if(i === index){
          track.scheduleStop();
      }
      return i !== index
    });

    this.adjustDuration();
    this.draw(this.render());
  }
```

**Comment by naomiaro** [comment]
https://naomiaro.github.io/waveform-playlist/stem-tracks.html

There's a remove track/collapse track option in  the newly released 4.0.0.
